src/ToastProvider.js

```diff
@@ -229,20 +245,35 @@ export class ToastProvider extends Component<Props, State> {
   }
 }
 
-export const ToastConsumer = ({ children }: { children: Context => Node }) => (
-  <Consumer>{context => children(context)}</Consumer>
-);
+export const ToastConsumer = ({
+  name = DEFAULT_CONTEXT_NAME,
```

Comment:
Each of the 3 ways of consuming toast context (`<ToastConsumer>`, `withToastManager()`, and `useToasts()`) now all accept an optional `name` prop.

That `name` is used to lookup a context created by a provider.

If the named context isn't known, then an error is thrown. Otherwise, that named context is the one used for the life of the component.

In this way, any deeply nested consumers can connect to any parent provider no matter how many they are as long as they have unique `name`s.